src/openms/source/ANALYSIS/OPENSWATH/DIAPrescoring.cpp

```diff
@@ -130,59 +131,85 @@ namespace OpenMS
+                                       Constants::C13C12_MASSDIFF_U,
+                                       chg);
+      DIAHelpers::addPreisotopeWeights(transition.getProductMZ(),
+                                       spectrumWIsoNegPreIso,
+                                       nrNegPeaks,
+                                       -totalNegWeight / (double(lt.size() * nrNegPeaks)),
+                                       Constants::C13C12_MASSDIFF_U,
+                                       chg);
+    }
+    //sort by mz
+    DIAHelpers::sortByFirst(spectrumWIso);
+    DIAHelpers::sortByFirst(spectrumWIsoNegPreIso);
+
+    // compare against the spectrum with 0 weight preIsotope peaks
+    std::vector<double> mzTheor, intTheor;
+    DIAHelpers::extractFirst(spectrumWIso, mzTheor);
+    DIAHelpers::extractSecond(spectrumWIso, intTheor);
     std::vector<double> intExp, mzExp;
-    DIAHelpers::integrateWindows(spec, theomasses, dia_extract_window_, intExp, mzExp);
+    DIAHelpers::integrateWindows(spec, mzTheor, dia_extract_window_, intExp, mzExp);
     std::transform(intExp.begin(), intExp.end(), intExp.begin(), OpenSwath::mySqrt());
-    std::transform(theorint.begin(), theorint.end(), theorint.begin(), OpenSwath::mySqrt());
-
-    double intExptotal = std::accumulate(intExp.begin(), intExp.end(), 0.0);
-    double intTheorTotal = std::accumulate(theorint.begin(), theorint.end(), 0.0);
-
-    OpenSwath::normalize(intExp, intExptotal, intExp);
-    OpenSwath::normalize(theorint, intTheorTotal, theorint);
+    std::transform(intTheor.begin(), intTheor.end(), intTheor.begin(), OpenSwath::mySqrt());
 
-    manhattan = OpenSwath::manhattanDist(intExp.begin(), intExp.end(), theorint.begin());
+    // get sum for normalization. All entries in both should be positive
+    double intExpTotal = std::accumulate(intExp.begin(), intExp.end(), 0.0);
+    double intTheorTotal = std::accumulate(intTheor.begin(), intTheor.end(), 0.0);
 
-    //std::cout << spectrum.size() << std::endl;
-    DIAHelpers::addPreisotopeWeights(firstIstotope, spectrum2, 2, -0.5);
-    std::vector<double>  theorint2;
-    DIAHelpers::extractSecond(spectrum, theorint2);
-    std::transform(theorint2.begin(), theorint2.end(), theorint2.begin(), OpenSwath::mySqrt());
+    OpenSwath::normalize(intExp, intExpTotal, intExp);
+    OpenSwath::normalize(intTheor, intTheorTotal, intTheor);
 
-    intExptotal = OpenSwath::norm(intExp.begin(), intExp.end());
-    intTheorTotal = OpenSwath::norm(theorint2.begin(), theorint2.end());
+    manhattan = OpenSwath::manhattanDist(intExp.begin(), intExp.end(), intTheor.begin());
 
-    OpenSwath::normalize(intExp, intExptotal, intExp);
-    OpenSwath::normalize(theorint2, intTheorTotal, theorint2);
+    // compare against the spectrum with negative weight preIsotope peaks
+    std::vector<double> intTheorNeg;
+    // WARNING: This was spectrumWIso and therefore with 0 preIso weights in earlier versions! Was this a bug?
```

Comment:
@grosenberger @hroest Do you know if it was intentional to not use negative weights for the preisotopes? E.g. after benchmarking you realized that it is penalizing too much and then just forgot to revert that part?